core/input.js

```diff
@@ -109,19 +109,21 @@ Blockly.Input.prototype.insertFieldAt = function(index, field, opt_name) {
   field.name = opt_name;
   field.setVisible(this.isVisible());
 
-  if (field.prefixField) {
+  var fieldDropdown = /** @type {Blockly.FieldDropdown} */ (field);
```

Comment:
I don't think it's safe to cast this to `FieldDropdown`.

Reply:
Not safe to cast the `field` var, but I don't see why its not safe to create a new var that is the cast of field as a `FieldDropdown`.

Reply:
You're casting so that you can access the field without the type checker warning, I assume. I guess it's technically safe but normally we cast when we already know the type and it's just that we've lost that information along the way, which isn't true here.

Reply:
Not quite, a type cast is not always an assertion that we know what the type is. In this case it's necessary to type cast because `prefixField` does not exist on type `Blockly.Field`, so from the compiler's perspective how can you say ``if (field.prefixField) `` if there is no such thing, instead what the cast is saying is assume that the field is a `Blockly.FieldDropdown` then proceed with that check as follows.

Perhaps the reason this seemed weird to you is because (you're right), we haven't made that assertion in the code. The above is a non-functional change, perhaps a cleaner change would be to say: 
- either cast it to an interface that implements prefixField, that way the code reads: any field that implements prefixField fits in here and should proceed in this code path, which is functionally what it's doing, but we're making assumptions here that only the DropdownField ever goes down this path.
- another functional alternative here is to check whether or not the field is an instanceof `Blockly.FieldDropdown` and only then proceed down the path, the compiler will implicitly cast the field to a `FieldDropdown` and "know" that prefixField exists.

The interface option here is overhead I feel, but an instanceof check would work. Only problem there is I don't think Blockly.FieldDropdown is one of the "required" fields, and so we can't assume that module exists, so we're back at "assuming".

Reply:
I see.  In that case, the cast is fine--it's not any worse than what we were doing before--and I'll keep this in mind as another corner.